empmanager/frontend2/src/components/form/elements/__test__/MainNavigation.test.js

```diff
@@ -0,0 +1,64 @@
+import { act, fireEvent, render, screen } from '@testing-library/react';
+import React from 'react';
+import { MainNavigation } from '../../../layout/MainNavigation';
+import { BrowserRouter, MemoryRouter } from 'react-router-dom';
+import userEvent from '@testing-library/user-event';
+import App from '../../../../App';
+import { RootStore } from '../../../../stores/root-store';
+import { RootStoreProvider } from '../../../../stores/root-store-provider';
+
+test('Menu displays on page correctly', () => {
+    const { getByTestId } = render(
+        <BrowserRouter>
+            <MainNavigation />
+        </BrowserRouter>,
+    );
+
+    const mainMenuEl = getByTestId('main-header');
+
+    expect(mainMenuEl).toBeInTheDocument();
+});
+
+test('Menu items display correctly in menu', () => {
+    const { getByTestId } = render(
+        <BrowserRouter>
+            <MainNavigation />
+        </BrowserRouter>,
+    );
+
+    const CompanyListEl = getByTestId('menu-company-list-item');
+
+    expect(CompanyListEl.textContent).toBe('Seznam firem');
+});
+
+test('Menu items link correctly redirects to right page', () => {
+    Object.defineProperty(window, 'matchMedia', {

```

Comment:
Why is this needed?

Reply:
Jest will crash without defining this property. According to their own documentation, they have not yet completed 
their implemented of a method Window.matchMedia() which is used in this test.
 https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom

I found this property definition to be their official solution, according to multiple threads.

Reply:
Ok :) In such "non-obvious" cases, I suggest to include a comment with a link that explains what is going on.

Reply:
Good idea, added explanation in comment